### ISSUE TITLE
Structured text link fails validation if target blank enabled

### ISSUE THREAD

**Issue by RomainFrancony** [voorhoede/datocms-plugin-translate-fields#40]
When inserting a link in a strutured text field, you can make the link open in a new tab. If you do so and then translate the field, you won't be able to save as the Dato validation will fails giving only the message "Invalid format".

That's because the link is stored like so in the original value : 
```json
[
  {
    "url": "https://example.org",
    "meta": [
      {
        "id": "target",
        "value": "_blank"
      }
    ]
  }
]
```

You can see the `target` attribute that was set by Dato inside a `meta` array. But then the translated field looks like that : 

```json
[
  {
    "url": "https://example.org",
    "meta": [
      {
        "value": "_blank"
      }
    ]
  }
]
```

As you can see the `id` for the target attribute is gone.

## Error source

The error come from here where the plugin recursively remove the `id` property: 
https://github.com/voorhoede/datocms-plugin-translate-fields/blob/71da761bc2971e51bac9262c2f1a32684a013b86/src/lib/translation.ts#L140

## Fix

The quick solution that I found is to not remove the property if the object's key is `meta` in [helpers.ts#L54](https://github.com/voorhoede/datocms-plugin-translate-fields/blob/71da761bc2971e51bac9262c2f1a32684a013b86/src/lib/helpers.ts#L54)

```js
if (key === 'meta' && Array.isArray(val)) {
  newObj[key] = val // Keep the original array without removing the property
} else if (!propertyNames.includes(key)) {
  newObj[key] = removePropertyRecursively(val, propertyNames)
}
```

I don't known if this is the most elegant solution but I can't think of something else for the moment.

**Comment by velomovies** [comment]
Hi @RomainFrancony,

Thanks for the clear explanation. I think this is a good solution, since the `meta` value is reserved and cannot be used. So it should never be a problem. I will look into this next week and add a robust solution.

**Comment by timvandevelde** [comment]
@velomovies did you manage to look into this? The plugin works great, but all of our content is structured text with external links so can't use it currently.

**Comment by velomovies** [comment]
@timvandevelde and @RomainFrancony, sorry for the wait! 

The `meta` tag will be skipped and thus keep the `id`. It will be available in `v1.11.0` when Dato picked it up.

If anything comes up, feel free to open a new issue.
